src/components/Exercise.tsx

```diff
@@ -78,6 +80,13 @@ export const Exercise = ({
 }: ExerciseWithStepDataProps | ExerciseWithQuestionStatesProps) => {
   const legacyStepRender = 'feedback_html' in step;
   const questionsRef = React.useRef<Array<HTMLDivElement>>([]);
+  const container = React.useRef<HTMLDivElement>(null);
+
+  const typesetExercise = () => {
+    if (container.current) {
+      typesetMath(container.current);
+    }
+  };
```

Comment:
you may want to wrap this in a React.useCallback, especially if you need to add it to the useEffect dependencies